lld/ELF/LinkerScript.cpp

```diff
@@ -717,14 +752,66 @@ void LinkerScript::processSectionCommands() {
       } else if (process(osec)) {
         osec->sectionIndex = i++;
       }
+    } else if (auto *sc = dyn_cast<SectionClassDesc>(base)) {
+      for (InputSectionDescription *isd : sc->sc.commands) {
+        isd->sectionBases =
+            computeInputSections(isd, ctx.inputSections, sc->sc);
+        for (InputSectionBase *s : isd->sectionBases) {
+          // Section classes with --enable-non-contiguous-regions may contain
```

Comment:
This comment is not accurate. This code path is useful even if --enable-non-contiguous-regions is unused.

Reply:
I think the comment was probably confusing; I've reworded it to hopefully be clearer. It should be the case that the only way a section can be assigned to the class *and* have a different parent than the class is using `--enable-non-contiguous-regions`; otherwise, having a parent would put it out of scope for matching a class.